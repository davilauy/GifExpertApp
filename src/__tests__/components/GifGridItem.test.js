import React from "react";
import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Prueba en: <GifGridItem />", () => {
  const titleTest = "Prueba";
  const urlTest =
    "https://media0.giphy.com/media/sEULHciNa7tUQ/giphy.gif?cid=ccfe99e00iom4cwxm6skphpn0b6h98tzsqqy3wgl7s8isrhc&rid=giphy.gif";

  const wrapper = shallow(<GifGridItem title={titleTest} url={urlTest} />);

  test("Debe de mostrar el contenido correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de renderizar un parrafo con el title ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(titleTest);
  });

  test("Debe de renderizar imagen con src y alt", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(urlTest);
    expect(img.prop("alt")).toBe(titleTest);
  });

  test("Debe de contener animacion animate__fadeIn ", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
