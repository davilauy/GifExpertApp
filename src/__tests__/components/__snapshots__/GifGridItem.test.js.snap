// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en: <GifGridItem /> Debe de mostrar el contenido correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Prueba"
    src="https://media0.giphy.com/media/sEULHciNa7tUQ/giphy.gif?cid=ccfe99e00iom4cwxm6skphpn0b6h98tzsqqy3wgl7s8isrhc&rid=giphy.gif"
    title="Prueba"
  />
  <p>
    Prueba
  </p>
</div>
`;
